packages/browser/src/Session.ts

```diff
@@ -132,16 +133,43 @@ function isLoggedIn(
 /**
  * A {@link Session} object represents a user's session on an application. The session holds state, as it stores information enabling acces to private resources after login for instance.
  */
-export class Session extends EventEmitter {
+export class Session
+  extends SessionEventEmitter
+  implements ISessionEventEmitter
+{
   /**
    * Information regarding the current session.
    */
   public readonly info: ISessionInfo;
 
+  /**
+   * Session attribute exposing the EventEmitter interface, to listen on session
+   * events such as login, logout, etc.
+   */
+  public readonly events: SessionEventEmitter;
+
   private clientAuthentication: ClientAuthentication;
 
   private tokenRequestInProgress = false;
 
+  /**
+   * Proxy handler to ease transition to the Session no longer being an EvenEmitter,
+   * and instead Session["events"] being a SessionEventEmitter. For the time being,
+   * Session["events"] is just a proxy to the SessionEventEmitter interface of
+   * the Session.
+   *
+   * When Session no longer implements SessionEventEmitter, this can be removed.
+   * @hidden
+   */
+  private handler = {
+    // This proxy is only a temporary measure until Session no longer extends
+    // SessionEventEmitter, and the proxying is no longer necessary.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    get(target: any, prop: any, receiver: any) {
+      return Reflect.get(target, prop, receiver);
+    },
+  };
```

Comment:
Since this will proxy every property/method of the `Session` it might be worth throwing an error here if consumers try and access properties or methods that are not available in the `EventEmitter` API.

Reply:
I agree, and that was actually part of the initial commit in 5d6228c02a04d262bb2eb1d8b0611f46669d7383. However, I reverted this in 91c49d4388d7ac8734be4942c3b1cf7622445fd7, because these checks are necessary when applied on the public API, but they also apply to the internal calls, which means the whole prototype chain should be present, as internal calls go beyond the `EventEmitter`. However, maybe reversing the logic, and throwing if a called method is in the `Session` API and not in the `EventEmitter` one (rather than just the latter condition) would work.

Reply:
I think da7b5dfc fixes the issue: any call to `session.events` that is part of `Session` but not of `EventEmitter` will throw, which prevents using the proxied `session.events` as the main `session`.